foundation-ui/auth/AuthService.js

```diff
@@ -0,0 +1,105 @@
+import Hooks from '../hooks';
+
+const hooks = Hooks();
+
+/**
+ * AuthService relies on a single instance of Hooks.
+ * It utilses the hooks actions to notify AuthPoints of changes and
+ * filters to allow external packages to register for and change the content
+ * output in the AuthPoint
+ *
+ * @example
+ * User of service would listen for updates and possibly change it
+ * import {AuthService} from 'foundation-ui/mount';
+ * AuthService.register('myID', function (value) {
+ *   return value.replace('my', 'myChanged');
+ * });
+ *
+ * AuthPoint would register itself with the AuthService to be able to get
+ * updates on content
+ * import {AuthService} from 'foundation-ui/mount';
+ * function updateContent() {
+ *   let newContent = AuthService.getContent(
+ *     'myID',
+ *     React.Children.toArray(children),
+ *     props
+ *   );
+ *
+ *   // Do something with newContent
+ * }
+ * AuthService.addListener('myID', updateContent);
+ */
+const AuthService = {
```

Comment:
We should agree an a pattern for creating "singleton" services, as we at the moment either create objects directly or return instances by using new within the module scope. But that should not block this PR.